File: TravelForecastBackend/src/main/java/com/travel/entity/common/EmergencyRescue.java
package com.travel.entity.common;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 紧急救援表
 */
@Data
@TableName("emergency_rescue")
public class EmergencyRescue implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 救援ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /**
     * 求救用户ID
     */
    private Long userId;

    /**
     * 景区ID
     */
    private Long scenicId;

    /**
     * 救援类型：MEDICAL(医疗),LOST(走失),ACCIDENT(事故),OTHER(其他)
     */
    private String rescueType;

    /**
     * 救援状态：PENDING(待处理),PROCESSING(处理中),COMPLETED(已完成),CANCELLED(已取消)
     */
    private String status;

    /**
     * 联系人姓名
     */
    private String contactName;

    /**
     * 联系电话
     */
    private String contactPhone;

    /**
     * 具体位置描述
     */
    private String location;

    /**
     * 纬度
     */
    private BigDecimal latitude;

    /**
     * 经度
     */
    private BigDecimal longitude;

    /**
     * 详细描述
     */
    private String description;

    /**
     * 紧急程度：URGENT(紧急),NORMAL(一般)
     */
    private String emergencyLevel;

    /**
     * 处理人ID
     */
    private Long handlerId;

    /**
     * 处理人姓名
     */
    private String handlerName;

    /**
     * 处理时间
     */
    private LocalDateTime handleTime;

    /**
     * 完成时间
     */
    private LocalDateTime completeTime;

    /**
     * 处理备注
     */
    private String handleNotes;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /**
     * 软删除时间 (NULL=未删除, 时间戳=已删除)
     */
    @TableLogic(value = "null", delval = "now()")
    private LocalDateTime deletedAt;
}

File: TravelForecastFrontend/web/src/api/emergencyRescue.ts
import request from '@/utils/request'

// 紧急救援相关接口 - 已修复API路径问题

export interface EmergencyRescue {
  id?: number
  userId?: number
  scenicId: number
  rescueType: string  // MEDICAL, LOST, ACCIDENT, OTHER
  status?: string  // PENDING, PROCESSING, COMPLETED, CANCELLED
  contactName: string
  contactPhone: string
  location: string
  latitude?: number
  longitude?: number
  description?: string
  emergencyLevel?: string  // URGENT, NORMAL
  handlerId?: number
  handlerName?: string
  handleTime?: string
  completeTime?: string
  handleNotes?: string
  createdAt?: string
  updatedAt?: string
}

/**
 * 创建救援请求（用户端）
 */
export const createRescue = (data: EmergencyRescue) => {
  return request({
    url: '/emergency-rescue/create',
    method: 'post',
    data
  })
}

/**
 * 获取用户的救援记录（用户端）
 */
export const getMyRescues = () => {
  return request({
    url: '/emergency-rescue/my-rescues',
    method: 'get'
  })
}

/**
 * 获取景区的救援列表（商家端）
 */
export const getRescuesByScenicId = (scenicId: number) => {
  return request({
    url: `/emergency-rescue/scenic/${scenicId}`,
    method: 'get'
  })
}

/**
 * 商家获取其管理的景区的所有救援列表（商家端）
 */
export const getMerchantRescues = () => {
  return request({
    url: '/emergency-rescue/merchant/list',
    method: 'get'
  })
}

/**
 * 处理救援请求（商家端）
 */
export const handleRescue = (rescueId: number, handlerName: string) => {
  return request({
    url: `/emergency-rescue/handle/${rescueId}`,
    method: 'post',
    data: { handlerName }
  })
}

/**
 * 完成救援请求（商家端）
 */
export const completeRescue = (rescueId: number, handleNotes: string) => {
  return request({
    url: `/emergency-rescue/complete/${rescueId}`,
    method: 'post',
    data: { handleNotes }
  })
}

/**
 * 取消救援请求
 */
export const cancelRescue = (rescueId: number) => {
  return request({
    url: `/emergency-rescue/cancel/${rescueId}`,
    method: 'post'
  })
}

/**
 * 获取景区救援统计信息（商家端）
 */
export const getRescueStats = (scenicId: number) => {
  return request({
    url: `/emergency-rescue/stats/${scenicId}`,
    method: 'get'
  })
}

/**
 * 获取救援详情
 */
export const getRescueDetail = (rescueId: number) => {
  return request({
    url: `/emergency-rescue/detail/${rescueId}`,
    method: 'get'
  })
}

/**
 * 删除救援记录（商家端）
 */
export const deleteRescue = (rescueId: number) => {
  return request({
    url: `/emergency-rescue/delete/${rescueId}`,
    method: 'delete'
  })
}